### PR TITLE
GuardDuty integration tests

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|-------------|
|      #3342     |

## Description

This PR adds tests for AWS GuardDuty integration

#### Tier 0
- Default configuration tests
- `remove_from_bucket` parameter tests
- `only_logs_after` parameter tests
- `path` parameter tests
- `discard_regex` parameter tests

#### Tier 1
- `only_logs_after` parameter tests
---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @nico-stefani  (Developer)  | `tests/integration/test_aws` | ⚫⚫⚫ | :green_circle: :red_circle:  | Ubuntu 22.04    |    | [^1][^2] |
| @reviewer (Reviewer)   | | ⚫⚫⚫ |  |   |      | Nothing to highlight |

### Tier 0 results
```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest -k 'guardduty' --tier 0
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 33 items / 17 deselected / 16 selected

test_basic.py ..                                                         [ 12%]
test_discard_regex.py ..                                                 [ 25%]
test_only_logs_after.py ....                                             [ 50%]
test_path.py ......                                                      [ 87%]
test_remove_from_bucket.py ..                                            [100%]

=============================== warnings summary ===============================
../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433
  /usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52
  /usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/stepwise
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========== 16 passed, 17 deselected, 2 warnings in 333.61s (0:05:33) ===========
```

### Tier 1 results

```python
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 33 items / 31 deselected / 2 selected

test_only_logs_after.py FF                                               [100%]

=================================== FAILURES ===================================
________ test_multiple_calls[guardduty_only_logs_after_multiple_calls] _________
------------------------------ Captured log call -------------------------------
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-guardduty-integration-tests', '--type', 'guardduty', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Table does not exist; create
DEBUG    wazuh_testing:cli_utils.py:52 The functionality to process GuardDuty logs stored in S3 via Kinesis was deprecated in 4.5. Consider configuring GuardDuty to store its findings directly in an S3 bucket instead. Check https://documentation.wazuh.com/current/amazon/services/supported-services/guardduty.html for more information. 
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2023/02/02
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
ERROR    wazuh_testing:cli_utils.py:62 Some logs may were processed
ERROR    wazuh_testing:cli_utils.py:63 Results found: 0
ERROR    wazuh_testing:cli_utils.py:64 Results expected: 1
_____ test_multiple_calls[native_guardduty_only_logs_after_multiple_calls] _____
------------------------------ Captured log call -------------------------------
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-native-guardduty-integration-tests', '--type', 'guardduty', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Table does not exist; create
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Working on 819751203818 - us-east-1
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: AWSLogs/819751203818/GuardDuty/us-east-1/2023/02/02
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ No logs to process in bucket: 819751203818/us-east-1
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-native-guardduty-integration-tests', '--type', 'guardduty', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Working on 819751203818 - us-east-1
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/20
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/21/ukkce94f-eb7e-4f45-b324-834a0d0cas43.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/23/465ce94f-eb7e-4f45-b324-834a0d0cas43.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/26/987ce94f-eb7e-4f45-b324-834a0d0ccd08.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-native-guardduty-integration-tests', '--type', 'guardduty', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Working on 819751203818 - us-east-1
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/20
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/21/ukkce94f-eb7e-4f45-b324-834a0d0cas43.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/23/465ce94f-eb7e-4f45-b324-834a0d0cas43.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/26/987ce94f-eb7e-4f45-b324-834a0d0ccd08.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-native-guardduty-integration-tests', '--type', 'guardduty', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-22']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Working on 819751203818 - us-east-1
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/22
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/23/465ce94f-eb7e-4f45-b324-834a0d0cas43.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: AWSLogs/819751203818/GuardDuty/us-east-1/2022/11/26/987ce94f-eb7e-4f45-b324-834a0d0ccd08.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-native-guardduty-integration-tests', '--type', 'guardduty', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2']'
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Working on 819751203818 - us-east-1
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: AWSLogs/819751203818/GuardDuty/us-east-1/2023/02/02
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: AWSLogs/819751203818/GuardDuty/us-east-1/2023/02/02/dcf07e79-60ab-4e93-a3ca-e4a5a9eda2e9.jsonl.gz
DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
ERROR    wazuh_testing:cli_utils.py:62 The AWS module did not use the correct marker
ERROR    wazuh_testing:cli_utils.py:63 Results found: 0
ERROR    wazuh_testing:cli_utils.py:64 Results expected: 1
=========================== short test summary info ============================
FAILED test_only_logs_after.py::test_multiple_calls[guardduty_only_logs_after_multiple_calls]
FAILED test_only_logs_after.py::test_multiple_calls[native_guardduty_only_logs_after_multiple_calls]
================ 2 failed, 31 deselected, 3 warnings in 55.90s =================

```
[^1]: Because the module is not showing the message about no logs to process, the *tier 1* test `guardduty_only_logs_after_multiple_calls` is falling. This should be fixed in his respective issue.
[^2]: Because the module is not showing the correct message about the marker, the *tier 1* test `native_guardduty_only_logs_after_multiple_calls` is falling. This should be fixed in his respective issue.

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 1]
```diff
@@ -94,12 +103,210 @@ def get_data_sample(self) -> dict:
...
+            data.append(
+                [
+                    "2", cons.RANDOM_ACCOUNT_ID, get_random_interface_id(), get_random_ip(), get_random_ip(),
+                    get_random_port(), get_random_port(), "6", "39", "4698", "1622505433", "1622505730", "ACCEPT", "OK"
+                ]
+            )
+        buffer = StringIO()
+        csv.writer(buffer, delimiter=" ").writerows(data)
+
+        return buffer.getvalue()
```
Comment (nico-stefani) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:144]: For the reviewer, please disregard this already has been fixed in https://github.com/wazuh/wazuh-qa/pull/3699/.

**Review by fdalmaup — CHANGES_REQUESTED** [review]
The Tier 0 test `test_discard_regex[guardduty_discard_regex]` fails with the following error message:
<details>
    
    root@ubuntu-focal:/home/vagrant/wazuh-qa/tests/integration/test_aws# pytest -k 'guardduty' --tier 0
    ====================================================================== test session starts ======================================================================
    platform linux -- Python 3.8.10, pytest-5.0.0, py-1.11.0, pluggy-0.13.1
    rootdir: /home/vagrant/wazuh-qa/tests/integration, inifile: pytest.ini
    plugins: html-2.0.1, metadata-2.0.4, testinfra-5.0.0
    collected 24 items / 16 deselected / 8 selected                                                                                                                 
    
    test_basic.py .                                                                                                                                           [ 12%]
    test_discard_regex.py F                                                                                                                                   [ 25%]
    test_only_logs_after.py ..                                                                                                                                [ 50%]
    test_path.py ...                                                                                                                                          [ 87%]
    test_remove_from_bucket.py .                                                                                                                              [100%]
    
    =========================================================================== FAILURES ============================================================================
    __________________________________________________________ test_discard_regex[guardduty_discard_regex] __________________________________________________________
    
    configuration = {'metadata': {'bucket_name': 'wazuh-guardduty-integration-tests', 'bucket_type': 'guardduty', 'description': 'GuardDut...elements': [{'disabled': {'value': 'no'}}, {'bucket': {'attributes': [...], 'elements': [...]}}], 'section': 'wodle'}]}
    metadata = {'bucket_name': 'wazuh-guardduty-integration-tests', 'bucket_type': 'guardduty', 'description': 'GuardDuty discard regex configurations', 'discard_field': 'partition', ...}
    load_wazuh_basic_configuration = None, set_wazuh_configuration = None, clean_s3_cloudtrail_db = None, configure_local_internal_options_function = None
    truncate_monitored_files = None, restart_wazuh_function = None, wazuh_log_monitor = <wazuh_testing.tools.monitoring.FileMonitor object at 0x7f20a644b5e0>
    
        @pytest.mark.tier(level=0)
        @pytest.mark.parametrize('configuration, metadata', zip(configurations, configuration_metadata), ids=case_ids)
        def test_discard_regex(
            configuration, metadata, load_wazuh_basic_configuration, set_wazuh_configuration, clean_s3_cloudtrail_db,
            configure_local_internal_options_function, truncate_monitored_files, restart_wazuh_function, wazuh_log_monitor,
        ):
            """
            description: Fetch logs excluding the ones that match with the regex.
            test_phases:
                - setup:
                    - Load Wazuh light configuration.
                    - Apply ossec.conf configuration changes according to the configuration template and use case.
                    - Apply custom settings in local_internal_options.conf.
                    - Truncate wazuh logs.
                    - Restart wazuh-manager service to apply configuration changes.
                - test:
                    - Check in the ossec.log that a line has appeared calling the module with correct parameters.
                    - Check the expected number of events were forwarded to analysisd, only logs stored in the bucket and skips
                      the ones that match with regex.
                    - Check the database was created and updated accordingly.
                - teardown:
                    - Truncate wazuh logs.
                    - Restore initial configuration, both ossec.conf and local_internal_options.conf.
                    - Delete the uploaded file
            wazuh_min_version: 4.5.0
            parameters:
                - configuration:
                    type: dict
                    brief: Get configurations from the module.
                - metadata:
                    type: dict
                    brief: Get metadata from the module.
                - load_wazuh_basic_configuration:
                    type: fixture
                    brief: Load basic wazuh configuration.
                - set_wazuh_configuration:
                    type: fixture
                    brief: Apply changes to the ossec.conf configuration.
                - clean_s3_cloudtrail_db:
                    type: fixture
                    brief: Delete the DB file before and after the test execution.
                - configure_local_internal_options_function:
                    type: fixture
                    brief: Apply changes to the local_internal_options.conf configuration.
                - truncate_monitored_files:
                    type: fixture
                    brief: Truncate wazuh logs.
                - restart_wazuh_daemon_function:
                    type: fixture
                    brief: Restart the wazuh service.
                - wazuh_log_monitor:
                    type: fixture
                    brief: Return a `ossec.log` monitor.
            assertions:
                - Check in the log that the module was called with correct parameters.
                - Check the expected number of events were forwarded to analysisd.
                - Check the database was created and updated accordingly.
            input_description:
                - The `configuration_discard_regex` file provides the module configuration for this test.
                - The `cases_discard_regex` file provides the test cases.
            """
            bucket_name = metadata['bucket_name']
            bucket_type = metadata['bucket_type']
            only_logs_after = metadata['only_logs_after']
            discard_field = metadata['discard_field']
            discard_regex = metadata['discard_regex']
            found_logs = metadata['found_logs']
            skipped_logs = metadata['skipped_logs']
        
            pattern = fr'.*The "{discard_regex}" regex found a match in the "{discard_field}" field. The event will be skipped.'
        
            parameters = [
                'wodles/aws/aws-s3',
                '--bucket', bucket_name,
                '--aws_profile', 'qa',
                '--only_logs_after', only_logs_after,
                '--discard-field', discard_field,
                '--discard-regex', discard_regex,
                '--type', bucket_type,
                '--debug', '2'
            ]
        
            # Check AWS module started
            wazuh_log_monitor.start(
                timeout=global_parameters.default_timeout,
                callback=event_monitor.callback_detect_aws_module_start,
                error_message='The AWS module did not start as expected',
            ).result()
        
            # Check command was called correctly
            wazuh_log_monitor.start(
                timeout=global_parameters.default_timeout,
                callback=event_monitor.callback_detect_aws_module_called(parameters),
                error_message='The AWS module was not called with the correct parameters',
            ).result()
        
    >       wazuh_log_monitor.start(
                timeout=T_20,
                callback=event_monitor.callback_detect_event_processed_or_skipped(pattern),
                error_message=(
                    'The AWS module did not show correct message about discard regex or ',
                    'did not process the expected amout of logs'
                ),
                accum_results=found_logs + skipped_logs
            ).result()
    
    test_discard_regex.py:130: 
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    /usr/local/lib/python3.8/dist-packages/wazuh_testing/tools/monitoring.py:201: in start
        self._result = monitor.start(timeout=timeout, callback=callback, accum_results=accum_results,
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    
    self = <wazuh_testing.tools.monitoring.QueueMonitor object at 0x7f20a644b670>, timeout = 20
    callback = <function callback_detect_event_processed_or_skipped.<locals>.<lambda> at 0x7f20a644edc0>, accum_results = 6, update_position = True
    timeout_extra = 0, error_message = ('The AWS module did not show correct message about discard regex or ', 'did not process the expected amout of logs')
    
        def start(self, timeout=-1, callback=_callback_default, accum_results=1, update_position=True, timeout_extra=0,
                  error_message=''):
            """Start the queue monitoring until the stop method is called."""
            if not self._continue:
                self._continue = True
                self._abort = False
                result = None
        
                while self._continue:
                    if self._abort:
                        self.stop()
                        if error_message:
                            logger.error(error_message)
                            logger.error(f"Results accumulated: "
                                         f"{len(result) if isinstance(result, list) else 0}")
                            logger.error(f"Results expected: {accum_results}")
    >                   raise TimeoutError(error_message)
    E                   TimeoutError: ('The AWS module did not show correct message about discard regex or ', 'did not process the expected amout of logs')
    
    /usr/local/lib/python3.8/dist-packages/wazuh_testing/tools/monitoring.py:469: TimeoutError
    --------------------------------------------------------------------- Captured stderr setup ---------------------------------------------------------------------
    2023-01-31 13:56:49,269 - wazuh_testing - DEBUG - Set local_internal_option to {'wazuh_modules.debug': '2', 'monitord.rotate_log': '0'}
    ---------------------------------------------------------------------- Captured log setup -----------------------------------------------------------------------
    DEBUG    wazuh_testing:conftest.py:619 Set local_internal_option to {'wazuh_modules.debug': '2', 'monitord.rotate_log': '0'}
    --------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------
    2023-01-31 13:57:23,337 - wazuh_testing - ERROR - ('The AWS module did not show correct message about discard regex or ', 'did not process the expected amout of logs')
    2023-01-31 13:57:23,337 - wazuh_testing - ERROR - Results accumulated: 4
    2023-01-31 13:57:23,337 - wazuh_testing - ERROR - Results expected: 6
    ----------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------
    ERROR    wazuh_testing:monitoring.py:465 ('The AWS module did not show correct message about discard regex or ', 'did not process the expected amout of logs')
    ERROR    wazuh_testing:monitoring.py:466 Results accumulated: 4
    ERROR    wazuh_testing:monitoring.py:468 Results expected: 6
    ------------------------------------------------------------------- Captured stderr teardown --------------------------------------------------------------------
    2023-01-31 13:57:26,520 - wazuh_testing - DEBUG - Restore local_internal_option to {'syscheck.debug': '2\n', 'monitord.rotate_log': '0\n'}
    --------------------------------------------------------------------- Captured log teardown ---------------------------------------------------------------------
    DEBUG    wazuh_testing:conftest.py:624 Restore local_internal_option to {'syscheck.debug': '2\n', 'monitord.rotate_log': '0\n'}
    ====================================================== 1 failed, 7 passed, 16 deselected in 186.87 seconds ======================================================

</details>

Additionally, the data generated for the testing corresponds to the deprecated implementation of AWSGuardDuty (since https://github.com/wazuh/wazuh/pull/15226). The tests should also address cases where native GuardDuty logs are used.

File: deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py [hunk 2]
```diff
@@ -62,6 +62,11 @@ def callback_detect_event_processed(line):
         return line
 
 
+def callback_detect_event_processed_or_skipped(pattern):
```
[the remark this answers is not in the document]
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py]: Fixed in b098d0d1c9598c76c8c0f67e8014652a683728d7

**Comment by nico-stefani** [comment]
> The Tier 0 test `test_discard_regex[guardduty_discard_regex]` fails with the following error message:
> 
> Additionally, the data generated for the testing corresponds to the deprecated implementation of AWSGuardDuty (since [wazuh/wazuh#15226](https://github.com/wazuh/wazuh/pull/15226)). The tests should also address cases where native GuardDuty logs are used.

The native test cases were added in cf8b22d5ad6cde58df808d5b1d8c1c443660193c. 
Also the test `guardduty_discard_regex` was fixed in 85f86e3ae03c5b143371e7a0e9317a8502946541.

**Review by fdalmaup — APPROVED** [review]
LGTM!

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**